src/ActivityStickyQueues/Program.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.Extensions.Logging;
+using Temporalio.Client;
+using Temporalio.Worker;
+using TemporalioSamples.ActivityStickyQueues;
+
+// Create a client to localhost on default namespace
+var client = await TemporalClient.ConnectAsync(new("localhost:7233")
+{
+    LoggerFactory = LoggerFactory.Create(builder =>
+        builder.
+            AddSimpleConsole(options => options.TimestampFormat = "[HH:mm:ss] ").
+            SetMinimumLevel(LogLevel.Information)),
+});
+
+async Task RunWorkerAsync()
+{
+    // Cancellation token cancelled on ctrl+c
+    using var tokenSource = new CancellationTokenSource();
+    Console.CancelKeyPress += (_, eventArgs) =>
+    {
+        tokenSource.Cancel();
+        eventArgs.Cancel = true;
+    };
+
+    var uniqueWorkerTaskQueue = Guid.NewGuid().ToString();
+
+    var nonStickyActivities = new NonStickyActivities(uniqueWorkerTaskQueue);
+
+    // Run worker until cancelled
+    Console.WriteLine("Running worker");
+
+    using var nonStickyWorker = new TemporalWorker(
+        client,
+        new TemporalWorkerOptions(taskQueue: "activity-sticky-queues-sample")
+            .AddActivity(nonStickyActivities.GetUniqueTaskQueue)
+            .AddWorkflow<FileProcessingWorkflow>());
+
+    using var stickyWorker = new TemporalWorker(
+        client,
+        new TemporalWorkerOptions(taskQueue: uniqueWorkerTaskQueue)
+            .AddActivity(StickyActivities.DownloadFileToWorkerFileSystemAsync)
+            .AddActivity(StickyActivities.CleanupFileFromWorkerFileSystemAsync)
+            .AddActivity(StickyActivities.WorkOnFileInWorkerFileSystemAsync));
+
+    try
+    {
+        await Task.WhenAll(nonStickyWorker.ExecuteAsync(tokenSource.Token), stickyWorker.ExecuteAsync(tokenSource.Token));
```

Comment:
The trouble with this is what if one worker threw a fatal exception not related to cancellation? Now it's just hung waiting on the other properly-working worker. I think this is fine for the sample, but something to think about.

Reply:
Curious how to handle this properly... Would you use `Task.WhenAny`?  Closest docs I can find is this https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/cancel-an-async-task-or-a-list-of-tasks

Reply:
It gets confusing (and it's not Temporal specific). It'd basically be a when any w/ captured non-cancel exception followed by a shutdown of each (or cancel of token) followed by a when all (maybe w/ exceptions ignored) followed by rethrowing what happened in when any.

Reply:
I tried a couple things to try and implement this, but TBH I'm struggling :)  I'm not sure how to cause either of the workers to blow up. 

This is what I tried: 

Note I added an `otherWorker` which throws after 10 seconds to test this out:

```csharp
async Task RunWorkerAsync()
{
    // Cancellation token cancelled on ctrl+c
    using var tokenSource = new CancellationTokenSource();
    Console.CancelKeyPress += (_, eventArgs) =>
    {
        tokenSource.Cancel();
        eventArgs.Cancel = true;
    };

    var uniqueWorkerTaskQueue = Guid.NewGuid().ToString();

    var nonStickyActivities = new NonStickyActivities(uniqueWorkerTaskQueue);

    // Run worker until cancelled
    Console.WriteLine("Running worker");

    using var nonStickyWorker = new TemporalWorker(
        client,
        new TemporalWorkerOptions(taskQueue: "activity-sticky-queues-sample")
            .AddActivity(nonStickyActivities.GetUniqueTaskQueue)
            .AddWorkflow<FileProcessingWorkflow>());

    using var stickyWorker = new TemporalWorker(
        client,
        new TemporalWorkerOptions(taskQueue: uniqueWorkerTaskQueue)
            .AddActivity(StickyActivities.DownloadFileToWorkerFileSystemAsync)
            .AddActivity(StickyActivities.CleanupFileFromWorkerFileSystemAsync)
            .AddActivity(StickyActivities.WorkOnFileInWorkerFileSystemAsync));

    using var otherWorker = new TemporalWorker(
        client,
        new TemporalWorkerOptions(taskQueue: "blah").AddActivity(nonStickyActivities.GetUniqueTaskQueue));

    var nonStickyWorkerTask = nonStickyWorker.ExecuteAsync(tokenSource.Token);
    var stickyWorkerTask = stickyWorker.ExecuteAsync(tokenSource.Token);
    var otherWorkerTask = otherWorker.ExecuteAsync(
        async () =>
        {
            await Task.Delay(TimeSpan.FromSeconds(10));
            throw new ArgumentException("Something bad happened");
        },
        tokenSource.Token);

    var task = await Task.WhenAny(nonStickyWorkerTask, stickyWorkerTask, otherWorkerTask);
    if (task.Exception is not null)
    {
        try
        {
            tokenSource.Cancel();
            await Task.WhenAll(nonStickyWorkerTask, stickyWorkerTask, otherWorkerTask);
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Unhandled ex {ex}");
            throw;
        }
    }
    else
    {
        Console.WriteLine("Worker cancelled");
    }
}
```

I the logs I see a warning `Worker finalization failed`

```
Running worker
[13:07:15] info: Temporalio.Activity:DownloadFileToWorkerFileSystem[0]
      Downloading https://temporal.io and saving to path C:\Users\jaken\AppData\Local\Temp\tmp92FC.tmp
[13:07:21] info: Temporalio.Activity:WorkOnFileInWorkerFileSystem[0]
      Did some work on C:\Users\jaken\AppData\Local\Temp\tmp92FC.tmp, checksum: 49d7419e6cba3575b3158f62d053f922aa08b23c64f05411cda3213b56c84ba4   
[13:07:22] info: Temporalio.Worker.TemporalWorker[0]
      User task completed, shutting down
[13:07:22] info: Temporalio.Worker.ActivityWorker[0]
      Beginning activity worker shutdown, will wait 00:00:00 before cancelling 0 activity instance(s)
[13:07:22] info: Temporalio.Worker.TemporalWorker[0]
      Worker cancelled, shutting down
[13:07:22] info: Temporalio.Worker.TemporalWorker[0]
      Worker cancelled, shutting down
[13:07:22] info: Temporalio.Worker.ActivityWorker[0]
      Beginning activity worker shutdown, will wait 00:00:00 before cancelling 1 activity instance(s)
[13:07:22] info: Temporalio.Worker.ActivityWorker[0]
      Beginning activity worker shutdown, will wait 00:00:00 before cancelling 0 activity instance(s)
[13:07:24] info: Temporalio.Activity:CleanupFileFromWorkerFileSystem[0]
      Removing C:\Users\jaken\AppData\Local\Temp\tmp92FC.tmp
[13:07:24] warn: Temporalio.Worker.TemporalWorker[0]
      Worker finalization failed
      System.InvalidOperationException: Cannot finalize, expected 1 reference, got 2
         at Temporalio.Bridge.Worker.FinalizeShutdownAsync()
         at Temporalio.Worker.TemporalWorker.ExecuteInternalAsync(Func`1 untilComplete, CancellationToken stoppingToken)
Unhandled ex System.ArgumentException: Something bad happened
   at Program.<>c.<<<Main>$>b__0_5>d.MoveNext() in C:\dev\temporalio\samples-dotnet\src\ActivityStickyQueues\Program.cs:line 55
--- End of stack trace from previous location ---
   at Temporalio.Worker.TemporalWorker.ExecuteInternalAsync(Func`1 untilComplete, CancellationToken stoppingToken)
   at Temporalio.Worker.TemporalWorker.ExecuteInternalAsync(Func`1 untilComplete, CancellationToken stoppingToken)
   at Program.<>c__DisplayClass0_0.<<<Main>$>g__RunWorkerAsync|0>d.MoveNext() in C:\dev\temporalio\samples-dotnet\src\ActivityStickyQueues\Program.cs:line 64
Unhandled exception. System.ArgumentException: Something bad happened
   at Program.<>c.<<<Main>$>b__0_5>d.MoveNext() in C:\dev\temporalio\samples-dotnet\src\ActivityStickyQueues\Program.cs:line 55
--- End of stack trace from previous location ---
   at Temporalio.Worker.TemporalWorker.ExecuteInternalAsync(Func`1 untilComplete, CancellationToken stoppingToken)
   at Temporalio.Worker.TemporalWorker.ExecuteInternalAsync(Func`1 untilComplete, CancellationToken stoppingToken)
   at Program.<>c__DisplayClass0_0.<<<Main>$>g__RunWorkerAsync|0>d.MoveNext() in C:\dev\temporalio\samples-dotnet\src\ActivityStickyQueues\Program.cs:line 64
--- End of stack trace from previous location ---
   at Program.<Main>$(String[] args) in C:\dev\temporalio\samples-dotnet\src\ActivityStickyQueues\Program.cs:line 89
   at Program.<Main>(String[] args)
```

Reply:
Yeah, that's just a warning and there may be a bug here on exception. We can file an issue, but I wouldn't worry about it for this sample.